backend-rust/migrations/0001_initialize.up.sql

```diff
@@ -247,6 +247,63 @@ CREATE TABLE bakers(
         BIGINT
 );
 
+-- Every WASM module on chain.
+CREATE TABLE smart_contract_modules(
+    index
+        BIGINT
+        PRIMARY KEY,
+    -- Module reference of the wasm module.
+    module_reference
+        CHAR(64)
+        UNIQUE
+        NOT NULL,
+    -- The absolute block height when the module was deployed.
+    deployment_block_height
+        BIGINT
+        NOT NULL,
+    -- Transaction index in the block deploying the module.
+    deployment_transaction_index
+        BIGINT
+        NOT NULL,
+    -- Embedded schema in the wasm module if present.
+    schema TEXT
+);
+
+-- Every contract instance on chain.
+CREATE TABLE contracts(
+    -- Index of the contract.
+    index
+        BIGINT
+        NOT NULL,
+    -- Sub index of the contract.
+    sub_index
+        BIGINT
+        NOT NULL,
+    -- Note: It might be better to use `module_reference_index` which would save storage space but would require more work in inserting/querying by the indexer.
+    -- Module reference of the wasm module.
+    module_reference
+        CHAR(64)
+        NOT NULL,
```

Comment:
For now, I decided to not include excessive `foreign_keys`/`references` constraints since they affect the sync time of the indexer. 

The initial focus will be to get everything coded to have a minimal backend working in Rust. The second focus will be on the queries that don't perform well (add indexes for that specific queries to speed them up). So we might add a key here but would like to see a query first that would benefit from it performance wise.